### DISCUSSION TITLE
error loading libxup.so specified in xrdp.ini, please add a valid entry like lib=libxrdp-vnc.so or similar

### DISCUSSION THREAD

**Discussion by myusrn** [neutrinolabs/xrdp#2732] [Q&A]
On an 2021 iMac M1 macOS venture 13.4 setup I built xrdp and associated binaries using [these](https://github.com/neutrinolabs/xrdp/wiki/Building-on-OSX-(not-official)) instructions using recent the xrdp-0.9.22.1.tar.gz and xorgxrdp-0.9.19.tar.gz downloads.  

I then established sharing | remote login ssh connections to macos m1 arm setup where i did this an ran `sudo /usr/local/sbin/xrdp -n --dump-config` and from a separate ssh session `sudo /usr/local/sbin/xrdp-sesman -n --dump-config`

When i attempted remote connection using remote desktop store app [ connection / mstse.exe desktop app ] i login and fail to connect with /var/log/xrdp.log showing the following /etc/xrdp/xrdp.ini lib=libxrdp-vnc.os related error.  

I tried editing `sudo nano /etc/xrdp/xrdp.ini` commenting out the [Xorg] sections `lib=libxup.so` and replacing it with `lib=libxrdp-vnc.so` and then restarting xrdp and xrdp-sesman processes. No change. 

Any insights on what i should do to try and fix this?

```
[20230615-13:35:19] [INFO ] sesman connect ok
[20230615-13:35:19] [INFO ] sending login info to session manager. Please wait...
[20230615-13:35:19] [INFO ] xrdp_wm_log_msg: login successful for user wendyqiu on display 10
[20230615-13:35:19] [INFO ] login successful for user wendyqiu on display 10
[20230615-13:35:19] [ERROR] xrdp_wm_log_msg: error loading libxup.so specified in xrdp.ini, please add a valid entry like lib=libxrdp-vnc.so or similar
[20230615-13:35:19] [INFO ] error loading libxup.so specified in xrdp.ini, please add a
[20230615-13:35:19] [INFO ] valid entry like lib=libxrdp-vnc.so or similar
[20230615-13:35:19] [ERROR] Failure setting up module
[20230615-13:35:56] [INFO ] xrdp_caps_process_pointer: client supports new(color) cursor
```

**Comment by firewave** [comment]
I have the same issues since moving from 0.9.x to 0.10.x.

**Comment by matt335672** [comment]
@firewave - based on your comment in #2732 this may not be the same thing. You're not using a Mac, which is pretty significant.

Can you open a separate issue on this, filling in all the installation details and providing your own logs? We have more Arch users than Mac users, but no-one else has mentioned this particular problem. Other AUR users have got v0.10.x working - see neutrinolabs/xorgxrdp#294

Thanks.

**Comment by firewave** [comment]
@matt335672 It seems this was an issue in the AUR package. There was a new version made available recently and after updating it is now working.

**Reply by myusrn** [reply]
Hi @firewave thanks for circling back with results of your additional experiements and findings on this matter.

q1. Does one have to do some special step to incorporate use of this new AUR package version you noted addresses macos apple silicon hosted xrdp service connection error detailed at start of this issue or does following the [building on macos](https://github.com/neutrinolabs/xrdp/wiki/Building-on-OSX-(not-official)) instructions using macos monterey 12.2.1 tested version [xrdp-0.9.19.tar.gz](https://github.com/neutrinolabs/xrdp/releases/download/v0.9.19/xrdp-0.9.19.tar.gz) [ or latest [xrdp-0.10.0-beta.3.tar.gz](https://github.com/neutrinolabs/xrdp/releases/download/v0.10.0-beta.3/xrdp-0.10.0-beta.3.tar.gz) ] + [xorgxrdp-v0.2.18.tar.gz](https://github.com/neutrinolabs/xorgxrdp/releases/tag/v0.2.18) [ or latest [xorgxrdp-v0.10.1.tar.gz](https://github.com/neutrinolabs/xorgxrdp/archive/refs/tags/v0.10.1.tar.gz) ] process to create macos apple silicon environment binaries take care of that for you?

q2. Have you been able to successfully get the [building on macos](https://github.com/neutrinolabs/xrdp/wiki/Building-on-OSX-(not-official)) denoted macos monterey 12.2.1 tested versions [xrdp-0.9.19.tar.gz](https://github.com/neutrinolabs/xrdp/releases/download/v0.9.19/xrdp-0.9.19.tar.gz) + [xorgxrdp-v0.2.18.tar.gz](https://github.com/neutrinolabs/xorgxrdp/releases/tag/v0.2.18) or latest versions [xrdp-0.10.0-beta.3.tar.gz](https://github.com/neutrinolabs/xrdp/releases/download/v0.10.0-beta.3/xrdp-0.10.0-beta.3.tar.gz) + [xorgxrdp-v0.10.1.tar.gz](https://github.com/neutrinolabs/xorgxrdp/archive/refs/tags/v0.10.1.tar.gz) based build successfully operating macos sonoma 14.4.1 apple silicon environment xrdp service connections?
